src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs

```diff
@@ -424,34 +433,55 @@ private static string GetBackupExtension(BackupType type, bool isFullBackup, boo
 
                     BackupTypeValidation();
 
+                    var currentLastEtag = _database.ReadLastEtag();
+                    var currentLastRaftIndexForBackup = GetDatabaseEtagForBackup();
+
                     if (_configuration.BackupType == BackupType.Backup ||
                         _configuration.BackupType == BackupType.Snapshot && _isFullBackup == false)
                     {
                         var backupType = _configuration.BackupType == BackupType.Snapshot ? "snapshot " : string.Empty;
-                        AddInfo($"Started an incremental {backupType}backup", onProgress);
+                        var backupSizeType = _isFullBackup ? "a full" : "an incremental";
+                        AddInfo($"Started {backupSizeType} {backupType}backup", onProgress);
 
                         // smuggler backup
                         var options = new DatabaseSmugglerOptionsServerSide
                         {
                             AuthorizationStatus = AuthorizationStatus.DatabaseAdmin,
                         };
                         if (_isFullBackup == false)
+                        {
                             options.OperateOnTypes |= DatabaseItemType.Tombstones;
+                            options.OperateOnTypes |= DatabaseItemType.CompareExchangeTombstones;
+                        }
+
+                        var currentBackupResult = CreateBackup(options, tempBackupFilePath, startDocumentEtag, startRaftIndex, onProgress);
 
-                        var lastEtagFromStorage = CreateBackup(options, tempBackupFilePath, startDocumentEtag, onProgress);
                         if (_isFullBackup)
-                            lastEtag = lastEtagFromStorage;
-                        else if (_database.ReadLastEtag() == _previousBackupStatus.LastEtag)
-                            lastEtag = startDocumentEtag ?? 0;
+                            internalBackupResult = currentBackupResult;
                         else
-                            lastEtag = _backupResult.GetLastEtag();
+                        {
+                            var docsNotChanged = currentLastEtag == _previousBackupStatus.LastEtag;
+                            var raftNotChanged = currentLastRaftIndexForBackup == _previousBackupStatus.LastRaftIndex.LastEtag;
+
+                            if (docsNotChanged && raftNotChanged)
+                            {
+                                internalBackupResult.LastDocumentEtag = startDocumentEtag ?? 0;
+                                internalBackupResult.LastRaftIndex = startRaftIndex ?? 0;
+                            }
+                            else
+                            {
+                                internalBackupResult.LastDocumentEtag = _backupResult.GetLastEtag();
+                                internalBackupResult.LastRaftIndex = _backupResult.GetLastRaftIndex();
```

Comment:
`currentLastEtag` vs `_backupResult.GetLastEtag()` - why is that?
Same for `_backupResult.GetLastRaftIndex()` vs `currentLastRaftIndexForBackup `

Reply:
done